design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,82 @@
+# Queries
+
+```graphql
+type Query {
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(
```

Comment:
How do we implement search across multiple entity types, like CMS, Category, Product? Don't we want to have unified search API?

Reply:
We discussed this in some other comments and on a few calls. The consensus IIRC was that we want it at some point in the future, but it's not an immediate need now. 

We thought that, for now, we'll be fine with type-specific searches, since those can still be batched into a single query.

```graphql
query SearchProductsAndCategories($query: String) {
   searchProducts(phrase: $query) {
      name
   }

   searchCategories(phrase: "$query" {
	  name
   }
}
```

This will not be suitable, however, in the case where we want the scoring/ranking of results to happen across types. But talking with @kokoc it sounded like this wasn't something we were ready to support in the near future.